### DISCUSSION TITLE
Hunk refinement does not work for unmerged files

### DISCUSSION THREAD

**Discussion by lmartelli** [magit/magit#5081] [Ideas]
magit version: 3.3.0

I have `magit-diff-refine-hunk` set to `all`, but it does not refine for unmerged files during rebase or merge.

**Comment by lmartelli** [comment]
It looks like Magit uses Emacs' diff here, so it's probably an issue/enhancement for Emacs.

**Reply by tarsius** [reply]
Yes, Magit relies on code that is part of Emacs for this.

I do not have any immediate plans to improve that part of Emacs myself or implement a replacement in Magit.
